### PR TITLE
build: Fix Qt installer build script.

### PR DESCRIPTION
**Pull request by Steveice10**
Few fixes for the Qt installer build script:
* Set the minimum CMake version to match our main requirement.
* The prefix output variable was removed from download_qt in favor of adding to the CMake prefix path. For this use case we need to match a glob though, so use get_external_prefix to fetch the prefix to a variable alone.
* The architecture parameter is not needed to download a tool, so don't try to resolve it or error if we can't.
* Add Tools to the path checked for when downloading, as we create the base directory already.

Tested on:
- [x] Windows
- [x] Linux
- [x] Mac

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6938)

### CODE REVIEW

File: CMakeModules/DownloadExternals.cmake [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
-                set(arch_path "msvc2019_arm64")
-            elseif ("x86_64" IN_LIST ARCHITECTURE)
-                set(arch_path "msvc2019_64")
+        if (NOT DOWNLOADING_QT_TOOL)
+            if (MINGW)
+                set(arch "win64_mingw")
+                set(arch_path "mingw_64")
+            elseif (MSVC)
+                if ("arm64" IN_LIST ARCHITECTURE)
+                    set(arch_path "msvc2019_arm64")
```
Comment (FearlessTobi) [inline CMakeModules/DownloadExternals.cmake:22]: is there a msvc2022 yet?

Comment (Steveice10) [reply CMakeModules/DownloadExternals.cmake:22]: Doesn’t look like it, should be compatible though.

For reference, this site lets you easily inspect the available downloads: https://ddalcino.github.io/aqt-list-server/

**Comment by vitor-k** [comment]
It's not super important but the latest version of aqt is 3.1.7 which includes having 6.5.1 and 6.5.2 as known versions

**Comment by Steveice10** [comment]
With the latest changes, TGP has confirmed it's working on Linux.

**Pull request merged**